src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineWindowsLauncher.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.google.jenkins.plugins.computeengine;
+
+import com.cloudbees.jenkins.plugins.sshcredentials.SSHAuthenticator;
+
+import com.cloudbees.plugins.credentials.common.StandardUsernameCredentials;
+import com.google.api.services.compute.model.AccessConfig;
+import com.google.api.services.compute.model.Instance;
+import com.google.api.services.compute.model.NetworkInterface;
+import com.google.api.services.compute.model.Operation;
+
+import com.trilead.ssh2.Connection;
+import com.trilead.ssh2.HTTPProxyData;
+import com.trilead.ssh2.SCPClient;
+import com.trilead.ssh2.ServerHostKeyVerifier;
+import com.trilead.ssh2.Session;
+import hudson.ProxyConfiguration;
+import hudson.model.TaskListener;
+import hudson.remoting.Channel;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import jenkins.model.Jenkins;
+
+public class ComputeEngineWindowsLauncher extends ComputeEngineComputerLauncher {
+    public final boolean useInternalAddress;
+
+    //TODO: make this configurable
+    public static final Integer SSH_PORT = 22;
+    public static final Integer SSH_TIMEOUT = 10000;
+    private static final Logger LOGGER = Logger.getLogger(ComputeEngineLinuxLauncher.class.getName());
+    private static int bootstrapAuthTries = 30;
+    private static int bootstrapAuthSleepMs = 15000;
+
+    public ComputeEngineWindowsLauncher(String cloudName, Operation insertOperation, boolean useInternalAddress) {
+        super(cloudName, insertOperation.getName(), insertOperation.getZone());
+        this.useInternalAddress = useInternalAddress;
+    }
+
+    protected void log(Level level, ComputeEngineComputer computer, TaskListener listener, String message) {
+        try {
+            ComputeEngineCloud cloud = computer.getCloud();
+            cloud.log(LOGGER, level, listener, message);
+        } catch (CloudNotFoundException cnfe) {
+            //TODO: figure out how to log without a handle to the cloud
```

Comment:
As a rule, we should never be swallowing exceptions like this, that leads to madness :) See the existing launcher's code as an example of what to do in this case: https://github.com/jenkinsci/google-compute-engine-plugin/blob/7bcfae34bd9f5493b0f99eb29bfe94ff6282e45a/src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineComputerLauncher.java#L70

Reply:
done